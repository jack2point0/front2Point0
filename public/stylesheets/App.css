* {
    font-family:"Open Sans","Helvetica Neue",Helvetica,Arial,sans-serif;
}

html {
  transition: 0.5s;
  background-color: #292f36;
}

a {
    text-decoration: none;
}

a:hover {
    color: #CF5300;
    text-decoration: none;
}

.form-control {
    border-radius: 5px;
}

.b {
  background-color: #292f36;
  min-height: 100vh;
  position: relative;
  transition: 0.5s
}

input,button {
  font-family: sans-serif;
  color: #292f36;
}

.table {
  display: flex;
  flex-direction: column;
  /* border: 2px solid black; */
  align-items: center;
  color: #EFE9E7;
}

.bodyContainer {
display: flex;
flex-direction: row;
justify-content: center;
/* border: 0.5px solid blue; */
max-width: 1220px;
flex-wrap: wrap;
}

.box {
  margin-top: 5px;
  margin-bottom: 5px;
  width: 255px;
}

input {
  width: 150px;
}
.testing {
  width: 80vw;
}

.tab-pane.fade {
  display: flex;
  flex-wrap: wrap;
}

.tabsContainer {
display: flex;
flex-direction: column;
max-width: 1300px;
flex-wrap: wrap;
justify-content: space-around;
}

.buttonDiv {
  display: flex;
  flex-direction: row;
  justify-content: center;
  justify-content: space-around;
  margin: 50px;
}

#notesField {
  margin-bottom: 40px;
}

.errors {
  color: #CF5300;
  font-weight: bold;
  padding-left: 120px;
}

.errors::first-letter {
  text-transform: capitalize;
}

.greeting {
  padding-left: 40px;
  margin-bottom: 20px;
  font-weight: 500;
  color: #CF5300;
  border-bottom: 1px solid #EFE9E7;
  margin-left: 50px;
  margin-right: 50px;
}


@media(max-width: 678px) {
    .tabsContainer {
        max-width: 405px;
    }
}
