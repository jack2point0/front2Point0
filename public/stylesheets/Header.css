.navbar {
  height: 70px;
  color: #EFE9E7;
  font-size: 16px;
  font-weight: bold;
  background-color: #0A090C;
  box-shadow: 0px 2px rgba(0, 0, 0, .15);
}

.logo {
  display: flex;
  align-items: center;
  background-image: url('/images/logo.png');
  background-size: contain;
  background-repeat: no-repeat;
  background-position: right;
  font-size: 26px;
  font-weight: bold;
  color: #EFE9E7;
  /* min-width: 274px; */
}

.logo:hover {
  color: #EFE9E7;
  text-decoration: none;
}










.logoContainer {
  float:left;
  padding-left: 20px;
  padding-top: 9px;
}

.smallLogoContainer {
    float:left;
    padding-left: 20px;
    padding-top: 9px
}

.navContainer {
  display: flex;
  float: right;
}

.navBarLink {
  height: 50px;
  margin-top: 12px;
  padding-top: 12px;
  padding-left: 12px;
  padding-right: 12px;
  margin-bottom: 15px;
  text-align: center;
  font-size: 12px;
}

.navBarLink::first-letter {
  font-size: 16px;
}

.navBarLink:hover {
  background-color: #242A30;
}

.navBarLink > a:hover {
  color: #EFE9E7;
  text-decoration: none;
}


.nav {
  color: #EFE9E7
}

/* mobile nav bar styling */
.sideNav {
  flex-direction: column;
  height: 100%;
  width: 200px;
  position: fixed;
  z-index: 9;
  top: 0;
  right: 0;
  background-color: black;
  overflow-x: hidden;
  padding-top: 70px;
  transition: 0.5s;
}

.sideNavLink {
  padding-top: 10px;
  padding-bottom: 10px;
  padding-left: 20px;
}

.sideNav a {
  padding: 0px 10px 10px 20px;
  text-decoration: none;
  font-size: 18px;
  display: flex;
  flex-direction: row;
  transition: 0.3s;
  color: #EFE9E7
}

.sideNav .btn-close {
  position: absolute;
  top: 0;
  right: 0;
  font-size: 42px;
}

.navBarLink > a:hover {
  color: #EFE9E7;
  text-decoration: none;
}

.fa-bars {
  margin-top: 15px;
  margin-right: 10px;
}

.fa-bars:hover {
  color: #EFE9E7;
  text-decoration: none;
}

.logoutLink, .homeLink {
  margin-right: 25px;
}

.sideNavLink {
    margin: 0px 5px 5px 10px;
    padding-top: 22px;
}

.sideNavLink:hover {
    background-color: #292F36;
}

.openSlide:hover {
  color: #EFE9E7;
  text-decoration: none;
}

#main {
  transition: margin-right 0.5s;
  padding: 20px;
  overflow: visible;
  width: 100%;
}

@media(max-width: 620px) {
  .navbar-nav {
    display: none;
  }
}

@media(min-width: 620px) {
  .openSlide {
    display: none;
  }
}

@media(max-width: 720px) {
    .smallLogoContainer {
        display: flex;
    }
    .logoContainer {
        display: none;
    }
}

@media(min-width: 720px) {
  .smallLogoContainer {
    display: none;
  }
  .logoContainer {
     display: flex;
  }
}

.container {
  margin: 0px;
  width: 100vw;
  display: flex;
}
