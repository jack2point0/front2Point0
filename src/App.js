import React, { Component } from 'react';
import Header from './Components/Header.js'
import Footer from './Components/Footer.js'
import Dashboard from './Pages/Dashboard.js'
import Home from './Pages/Home.js'
import Instructions from './Pages/Instructions.js'
import Registration from './Pages/Registration.js'
import Sign_In from './Pages/Sign_in.js'
import Task_Index from './Pages/Task_index.js'
import { Redirect, BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import AuthService from './services'
import { getMyTasks } from './API';
import Appointments from './Pages/Appointments'
import AptForm from './Pages/aptForm'
import EditMyTaskCard from './Components/editMyTaskCard'
import Profile from './Pages/Profile'


class App extends Component {
    constructor(props) {
      super(props)

      this.auth = new AuthService()
      this.state = {
        marginRight: 0,
        authenticated: this.auth.loggedIn(),
        hasToken: false,
        myTasks: [],
        userID: ''
      }
    }

  render() {
    return (
      <div style={{marginRight: this.state.marginRight}} className="b">
        <Header openSideMenu={this.openSideMenu}/>
        <div className="b">
          <Router>
            {
              (this.auth.loggedIn()) ?
              <Switch> //protected paths
                <Redirect from="/register" to="/" />
                <Redirect from="/login" to="/" />
                <Route exact path='/user/my_tasks/:id/edit' render={(routeProps) => <EditMyTaskCard refresh={this.refresh} {...routeProps}/>} />
                <Route exact path='/instructions' component={Instructions} />
                <Route exact path='/profile' render={(routeProps) => <Profile refresh={this.refresh} {...routeProps}/>} />
                <Route exact path='/tasks' render={(props) => <Task_Index userID={this.state.userID} />} />
                <Route exact path='/appointments' component={Appointments} />
                <Route exact path='/appointments/new' component={AptForm} />
                <Route path='/' render={(props) => <Dashboard myTasks={this.state.myTasks} />} />
              </Switch>

              :
              <Switch> //public paths
                <Route exact path='/login' render={(props) => <Sign_In refresh={this.refresh}/>} />
                <Route exact path='/register' render={(props) => <Registration refresh={this.refresh}/> }/>
                <Route path='/' component={Home} />
              </Switch>
            }
          </Router>
        </div>
          <footer >
            <Footer />
          </footer>
      </div>
    );
  }



  componentDidMount = () => {
    let thisUserID = this.auth.getUserId()
    if (thisUserID !== null && thisUserID !== undefined && thisUserID.length > 0) {
      getMyTasks(thisUserID)
      .then((APImyTasks)=> {
      // let { myTasks } = this.state
        this.setState({
          myTasks: APImyTasks,
          userID: thisUserID
        })
      })
    }
  }

  openSideMenu = () => {
    console.log("made it in");
    let { marginRight } = this.state
    marginRight = marginRight === 0 ? 200 : 0
    this.setState({
      marginRight: marginRight
    })
  }

  refresh = () => {
    this.setState ({
      authenticated: this.auth.loggedIn()
    })
  }
}

export default App;
