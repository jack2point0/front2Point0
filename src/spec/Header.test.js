import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import Header from '../Components/Header';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

const headerAtts = { width: 0 }

test ('renders the Header correctly', () => {
 const json = renderer
             .create(<Header header={headerAtts} />)
             .toJSON();
 expect(json).toMatchSnapshot();
})
