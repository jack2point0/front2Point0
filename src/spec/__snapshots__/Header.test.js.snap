// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header correctly 1`] = `
<main
  className="headerDiv navbar"
>
  <section
    className="logoContainer"
  >
    <div
      className="BrandName"
    >
      <a
        className="logo"
        href="/"
      >
        <i
          className="fas fa-arrow-alt-circle-up fa-2x"
        />
        wn-Up Grown-Up
      </a>
    </div>
  </section>
  <nav>
    <div>
      <div
        className="navContainer openSlide openSlideMenu"
        style={
          Object {
            "marginRight": 10,
          }
        }
      >
        <a
          href="#"
          onClick={[Function]}
        >
          <i
            class="fas fa-bars fa-2x"
          />
        </a>
      </div>
      <section
        className="nav navContainer navbar-nav"
      >
        <div
          className="navBarLink homeLink"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </div>
      </section>
    </div>
    <div
      className=""
      id="side-menu"
    >
      <section
        className="navContainer sideNav"
        style={
          Object {
            "width": 0,
          }
        }
      >
        <a
          className="btn-close"
          href="#"
          onClick={[Function]}
        >
          Ã—
        </a>
        <div
          className="sideNavLink"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </div>
      </section>
    </div>
  </nav>
</main>
`;
